apptests/test_generators.py

```diff
@@ -0,0 +1,326 @@
+        self.expected_keys = [
+            "videoId",
+            "channelId",
+            "channelTitle",
+            "publishTime",
+            "title",
+        ]
+        return super().setUp()
+
+    def test_valid_params(self):
+        response = gen.results_parser(example_response)
+        self.assertIsNotNone(response.get("items"))
+
+        self.assertIsNone(response.get("kind"))
+        self.assertIsNone(response.get("pageInfo"))
+        for key in self.expected_keys:
+            with self.subTest(key=key):
+                self.assertIn(key, response.get("items")[0])
+                self.assertIsNotNone(response.get("items")[0].get(key))
+
+    # it should check if results have items key
+    # def test_empty_results_param(self):
+    #     self.assertRaises(KeyError, gen.results_parser({}))
+
+
+class Test_add_video_details(unittest.TestCase):
+    def setUp(self) -> None:
+        self.ok_parsed_example = deepcopy(parsed_example)
+        self.nok_parsed_example = deepcopy(parsed_example)
+        return super().setUp()
+
+    def test_valid_video_list(self):
+        gen.add_video_details(self.ok_parsed_example)
+        video = self.ok_parsed_example[0]
+        self.assertIsNotNone(video.get("duration"))
+        self.assertIsNotNone(video.get("defaultAudioLanguage"))
+
+    # it should check if videoId key exists
+    # def test_invalid_video_list(self):
+    #     self.nok_parsed_example[0].pop("videoId")
+    #     gen.add_video_details(self.nok_parsed_example)
+    #     video = self.nok_parsed_example[0]
+    #     self.assertIsNone(video.get("duration"))
+    #     self.assertIsNone(video.get("defaultAudioLanguage"))
+
+
+class Test_add_transcripts_info(unittest.TestCase):
+    def setUp(self) -> None:
+        self.ok_parsed_example = deepcopy(parsed_example)
+        self.nok1_parsed_example = deepcopy(parsed_example)
+        self.nok2_parsed_example = deepcopy(parsed_example)
+        return super().setUp()
+
+    def test_valid_video_list(self):
+        gen.add_transcripts_info(self.ok_parsed_example)
+        video = self.ok_parsed_example[0]
+        self.assertIsNotNone(video.get("manuallyCreatedTranscripts"))
+        self.assertIsNotNone(video.get("generatedTranscripts"))
+
+    def test_invalid_video_list(self):
+        self.nok2_parsed_example[0].pop("title")
+        gen.add_transcripts_info(self.nok2_parsed_example)
+        video = self.nok2_parsed_example[0]
+        self.assertIsNotNone(video.get("manuallyCreatedTranscripts"))
+        self.assertIsNotNone(video.get("generatedTranscripts"))
+
+    # it should check if videoId key exists
+    # def test_lack_of_videoid(self):
+    #     self.nok1_parsed_example[0].pop("videoId")
+    #     gen.add_transcripts_info(self.nok1_parsed_example)
+    #     video = self.nok1_parsed_example[0]
+    #     self.assertIsNone(video.get("manuallyCreatedTranscripts"))
+    #     self.assertIsNone(video.get("generatedTranscripts"))
```

Comment:
If something goes wrong the `manuallyCreatedTranscripts` and `generatedTranscripts` should be a empty list.